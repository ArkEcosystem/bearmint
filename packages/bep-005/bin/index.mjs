#!/usr/bin/env node
import { join } from 'node:path'

import { globby } from 'globby'
import meow from 'meow'

process.env.NODE_ENV = 'test'

const { flags } = meow({
	flags: {
		config: {
			type: 'string',
		},
	},
	importMeta: import.meta,
})

const { files } = (await import(join(process.cwd(), flags.config))).default
const tests = (await globby(files)).map((path) => join(process.cwd(), path))

for (const test of tests) {
	await import(test)
}
