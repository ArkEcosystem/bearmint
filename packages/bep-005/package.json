{
	"name": "@bearmint/bep-005",
	"version": "0.0.0",
	"private": true,
	"description": "This library is the official reference implementation for BEP-005 (https://bep.dev/005)",
	"keywords": [
		"bearmint",
		"tendermint",
		"blockchain",
		"framework",
		"modular"
	],
	"homepage": "https://github.com/bearmint/bearmint",
	"repository": {
		"type": "git",
		"url": "git+https://github.com/bearmint/bearmint.git"
	},
	"funding": "https://funding.bearmint.com/",
	"license": "MIT",
	"author": "Bearmint Mining Company <hello@bearmint.com> (https://bearmint.com/)",
	"type": "module",
	"exports": "./distribution/index.js",
	"types": "./distribution/index.d.ts",
	"bin": {
		"roadrunner": "./bin/index.mjs"
	},
	"files": [
		"bin",
		"distribution"
	],
	"scripts": {
		"audit": "pnpm audit --fix --prod",
		"build": "pnpm run clean && tsc",
		"build:watch": "pnpm run clean && tsc -w",
		"clean": "rm -rf distribution",
		"depcheck": "depcheck --ignores=@bearmint/bep-004,@bearmint/bep-005,@bearmint/bep-031,@bearmint/bep-039,globby,meow",
		"deps:major": "ncu -u --target=latest",
		"deps:minor": "ncu -u --target=minor",
		"deps:patch": "ncu -u --target=patch"
	},
	"dependencies": {
		"@bearmint/bep-013": "workspace:^",
		"@bearmint/bep-018": "workspace:^",
		"@bearmint/bep-116": "workspace:^",
		"@sinonjs/fake-timers": "9.1.2",
		"ansi-styles": "6.2.1",
		"concordance": "5.0.4",
		"globby": "13.1.2",
		"lodash.get": "4.4.2",
		"lodash.has": "4.5.2",
		"meow": "^11.0.0",
		"nock": "13.2.9",
		"pretty-ms": "8.0.0",
		"sinon": "14.0.1",
		"strip-ansi": "7.0.1",
		"vitest": "^0.24.5"
	},
	"devDependencies": {
		"@types/node": "18.11.9",
		"@types/sinon": "10.0.13",
		"@types/sinonjs__fake-timers": "8.1.2"
	},
	"engines": {
		"node": ">=18",
		"pnpm": ">=7.9.0"
	},
	"publishConfig": {
		"access": "restricted"
	}
}
