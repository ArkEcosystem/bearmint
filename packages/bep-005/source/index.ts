export * from './expect.js'
export * from './fake.js'
export * from './nock.js'
export * from './spy.js'
export * from './stub.js'
