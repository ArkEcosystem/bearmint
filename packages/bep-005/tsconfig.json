{
	"compilerOptions": {
		"typeRoots": ["node_modules/@types"],
		"outDir": "distribution"
	},
	"exclude": ["node_modules", "test/**/**.ts"],
	"extends": "../../config/tsconfig.json",
	"include": ["source/**/**.ts"]
}
