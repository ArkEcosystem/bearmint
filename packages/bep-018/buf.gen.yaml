version: v1
plugins:
  - name: es
    out: source
    opt: target=ts
  - name: connect-web
    out: source
    opt: target=ts
