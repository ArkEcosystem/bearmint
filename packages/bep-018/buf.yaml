build:
  roots:
    - proto
breaking:
  use:
    - FILE
