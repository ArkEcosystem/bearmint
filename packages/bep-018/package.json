{
	"name": "@bearmint/bep-018",
	"version": "0.0.0",
	"description": "This library is the official reference implementation for BEP-018 (https://bep.dev/018)",
	"keywords": [
		"bearmint",
		"tendermint",
		"blockchain",
		"framework",
		"modular"
	],
	"homepage": "https://github.com/bearmint/bearmint",
	"repository": {
		"type": "git",
		"url": "git+https://github.com/bearmint/bearmint.git"
	},
	"funding": "https://funding.bearmint.com/",
	"license": "MIT",
	"author": "Bearmint Mining Company <hello@bearmint.com> (https://bearmint.com/)",
	"type": "module",
	"exports": "./distribution/index.js",
	"types": "./distribution/index.d.ts",
	"files": [
		"distribution"
	],
	"scripts": {
		"audit": "pnpm audit --fix --prod",
		"build": "pnpm run clean && pnpm run proto && tsc",
		"build:watch": "pnpm run clean && pnpm run proto && tsc -w",
		"clean": "rm -rf distribution",
		"depcheck": "depcheck --ignores=@bearmint/bep-004,@bearmint/bep-005,@bearmint/bep-031,@bearmint/bep-039,@bufbuild/protoc-gen-es",
		"deps:major": "ncu -u --target=latest",
		"deps:minor": "ncu -u --target=minor",
		"deps:patch": "ncu -u --target=patch",
		"docs": "typedoc source",
		"knip": "knip",
		"lint": "eslint 'source/**/*.ts' --format=codeframe --config=../../config/eslintrc.js --fix",
		"proto": "buf generate",
		"proto:format": "buf format -w"
	},
	"dependencies": {
		"@bufbuild/connect-web": "^0.3.0",
		"@bufbuild/protobuf": "0.2.0"
	},
	"devDependencies": {
		"@bufbuild/protoc-gen-connect-web": "^0.3.0",
		"@bufbuild/protoc-gen-es": "0.2.0"
	},
	"engines": {
		"node": ">=18",
		"pnpm": ">=7.9.0"
	},
	"publishConfig": {
		"access": "public"
	}
}
