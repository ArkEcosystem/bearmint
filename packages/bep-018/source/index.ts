export * from './account.js'
export * from './account_pb.js'
export * from './state_pb.js'
export * from './tendermint/index.js'
export * from './tx.js'
export * from './tx_pb.js'
export * from './types.js'
export type { AnyMessage, PartialMessage, PlainMessage } from '@bufbuild/protobuf'
export { Message } from '@bufbuild/protobuf'
