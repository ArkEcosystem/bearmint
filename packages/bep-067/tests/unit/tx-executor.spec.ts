import { expect, describe, it, beforeEach } from 'vitest'
import { extendExpect } from '@bearmint/bep-005'
extendExpect(expect)
import type { TxTestContext } from '@bearmint/bep-006'
import { BEP67MsgOp } from '@bearmint/bep-018'

import { createTx, setUp } from '../setup.js'

describe<TxTestContext>('Tx Executor', () => {
	setUp(beforeEach)

	it('should attach [base32cid] to the sender', async (context) => {
		const { primaryKey, sender, tx } = await createTx(context, {
			ops: [
				{
					type: 'base32cid',
					value: 'bafybeie5gq4jxvzmsym6hjlwxej4rwdoxt7wadqvmmwbqi7r27fclha2va',
				},
			],
		})

		await expect(await context.execute(tx)).toPassExecution()

		expect(
			await context.state.getMultiStore().get('@bearmint/bep-067').getListHex(sender.address),
		).toStrictEqual([primaryKey])

		expect(
			BEP67MsgOp.fromBinary(
				await context.state.getMultiStore().get('@bearmint/bep-067').get(primaryKey),
			).toJson(),
		).toMatchSnapshot()
	})

	it('should attach [cid] to the sender', async (context) => {
		const { primaryKey, sender, tx } = await createTx(context, {
			ops: [
				{
					type: 'cid',
					value: 'bafybeiasb5vpmaounyilfuxbd3lryvosl4yefqrfahsb2esg46q6tu6y5q',
				},
			],
		})

		await expect(await context.execute(tx)).toPassExecution()

		expect(
			await context.state.getMultiStore().get('@bearmint/bep-067').getListHex(sender.address),
		).toStrictEqual([primaryKey])

		expect(
			BEP67MsgOp.fromBinary(
				await context.state.getMultiStore().get('@bearmint/bep-067').get(primaryKey),
			).toJson(),
		).toMatchSnapshot()
	})

	it('should attach [cid] to the sender', async (context) => {
		const { primaryKey, sender, tx } = await createTx(context, {
			ops: [
				{
					type: 'cid',
					value: 'QmYjtig7VJQ6XsnUjqqJvj7QaMcCAwtrgNdahSiFofrE7o',
				},
			],
		})

		await expect(await context.execute(tx)).toPassExecution()

		expect(
			await context.state.getMultiStore().get('@bearmint/bep-067').getListHex(sender.address),
		).toStrictEqual([primaryKey])

		expect(
			BEP67MsgOp.fromBinary(
				await context.state.getMultiStore().get('@bearmint/bep-067').get(primaryKey),
			).toJson(),
		).toMatchSnapshot()
	})

	it('should attach [cid] to the sender', async (context) => {
		const { primaryKey, sender, tx } = await createTx(context, {
			ops: [
				{
					type: 'cid',
					value: 'zdj7WWeQ43G6JJvLWQWZpyHuAMq6uYWRjkBXFad11vE2LHhQ7',
				},
			],
		})

		await expect(await context.execute(tx)).toPassExecution()

		expect(
			await context.state.getMultiStore().get('@bearmint/bep-067').getListHex(sender.address),
		).toStrictEqual([primaryKey])

		expect(
			BEP67MsgOp.fromBinary(
				await context.state.getMultiStore().get('@bearmint/bep-067').get(primaryKey),
			).toJson(),
		).toMatchSnapshot()
	})

	it('should attach [cidPath] to the sender', async (context) => {
		const { primaryKey, sender, tx } = await createTx(context, {
			ops: [
				{
					type: 'cidPath',
					value: 'QmYjtig7VJQ6XsnUjqqJvj7QaMcCAwtrgNdahSiFofrE7o/path/to/file',
				},
			],
		})

		await expect(await context.execute(tx)).toPassExecution()

		expect(
			await context.state.getMultiStore().get('@bearmint/bep-067').getListHex(sender.address),
		).toStrictEqual([primaryKey])

		expect(
			BEP67MsgOp.fromBinary(
				await context.state.getMultiStore().get('@bearmint/bep-067').get(primaryKey),
			).toJson(),
		).toMatchSnapshot()
	})

	it('should attach [ipfsPath] to the sender', async (context) => {
		const { primaryKey, sender, tx } = await createTx(context, {
			ops: [
				{
					type: 'ipfsPath',
					value: '/ipfs/QmYjtig7VJQ6XsnUjqqJvj7QaMcCAwtrgNdahSiFofrE7o',
				},
			],
		})

		await expect(await context.execute(tx)).toPassExecution()

		expect(
			await context.state.getMultiStore().get('@bearmint/bep-067').getListHex(sender.address),
		).toStrictEqual([primaryKey])

		expect(
			BEP67MsgOp.fromBinary(
				await context.state.getMultiStore().get('@bearmint/bep-067').get(primaryKey),
			).toJson(),
		).toMatchSnapshot()
	})

	it('should attach [ipfsSubdomain] to the sender', async (context) => {
		const { primaryKey, sender, tx } = await createTx(context, {
			ops: [
				{
					type: 'ipfsSubdomain',
					value:
						'http://bafybeie5gq4jxvzmsym6hjlwxej4rwdoxt7wadqvmmwbqi7r27fclha2va.ipfs.dweb.link',
				},
			],
		})

		await expect(await context.execute(tx)).toPassExecution()

		expect(
			await context.state.getMultiStore().get('@bearmint/bep-067').getListHex(sender.address),
		).toStrictEqual([primaryKey])

		expect(
			BEP67MsgOp.fromBinary(
				await context.state.getMultiStore().get('@bearmint/bep-067').get(primaryKey),
			).toJson(),
		).toMatchSnapshot()
	})

	it('should attach [ipfsUrl] to the sender', async (context) => {
		const { primaryKey, sender, tx } = await createTx(context, {
			ops: [
				{
					type: 'ipfsUrl',
					value: 'https://ipfs.io/ipfs/QmYjtig7VJQ6XsnUjqqJvj7QaMcCAwtrgNdahSiFofrE7o',
				},
			],
		})

		await expect(await context.execute(tx)).toPassExecution()

		expect(
			await context.state.getMultiStore().get('@bearmint/bep-067').getListHex(sender.address),
		).toStrictEqual([primaryKey])

		expect(
			BEP67MsgOp.fromBinary(
				await context.state.getMultiStore().get('@bearmint/bep-067').get(primaryKey),
			).toJson(),
		).toMatchSnapshot()
	})

	it('should attach [ipnsPath] to the sender', async (context) => {
		const { primaryKey, sender, tx } = await createTx(context, {
			ops: [
				{
					type: 'ipnsPath',
					value: '/ipns/github.com',
				},
			],
		})

		await expect(await context.execute(tx)).toPassExecution()

		expect(
			await context.state.getMultiStore().get('@bearmint/bep-067').getListHex(sender.address),
		).toStrictEqual([primaryKey])

		expect(
			BEP67MsgOp.fromBinary(
				await context.state.getMultiStore().get('@bearmint/bep-067').get(primaryKey),
			).toJson(),
		).toMatchSnapshot()
	})

	it('should attach [ipnsSubdomain] to the sender', async (context) => {
		const { primaryKey, sender, tx } = await createTx(context, {
			ops: [
				{
					type: 'ipnsSubdomain',
					value:
						'http://bafybeiabc2xofh6tdi6vutusorpumwcikw3hf3st4ecjugo6j52f6xwc6q.ipns.dweb.link',
				},
			],
		})

		await expect(await context.execute(tx)).toPassExecution()

		expect(
			await context.state.getMultiStore().get('@bearmint/bep-067').getListHex(sender.address),
		).toStrictEqual([primaryKey])

		expect(
			BEP67MsgOp.fromBinary(
				await context.state.getMultiStore().get('@bearmint/bep-067').get(primaryKey),
			).toJson(),
		).toMatchSnapshot()
	})

	it('should attach [ipnsUrl] to the sender', async (context) => {
		const { primaryKey, sender, tx } = await createTx(context, {
			ops: [
				{
					type: 'ipnsUrl',
					value: 'https://ipfs.io/ipns/github.com',
				},
			],
		})

		await expect(await context.execute(tx)).toPassExecution()

		expect(
			await context.state.getMultiStore().get('@bearmint/bep-067').getListHex(sender.address),
		).toStrictEqual([primaryKey])

		expect(
			BEP67MsgOp.fromBinary(
				await context.state.getMultiStore().get('@bearmint/bep-067').get(primaryKey),
			).toJson(),
		).toMatchSnapshot()
	})

	it('should attach [multiaddr] to the sender', async (context) => {
		const { primaryKey, sender, tx } = await createTx(context, {
			ops: [
				{
					type: 'multiaddr',
					value: '/ip4/127.0.0.1/udp/1234',
				},
			],
		})

		await expect(await context.execute(tx)).toPassExecution()

		expect(
			await context.state.getMultiStore().get('@bearmint/bep-067').getListHex(sender.address),
		).toStrictEqual([primaryKey])

		expect(
			BEP67MsgOp.fromBinary(
				await context.state.getMultiStore().get('@bearmint/bep-067').get(primaryKey),
			).toJson(),
		).toMatchSnapshot()
	})

	it('should attach [multihash] to the sender', async (context) => {
		const { primaryKey, sender, tx } = await createTx(context, {
			ops: [
				{
					type: 'multihash',
					value: 'QmYjtig7VJQ6XsnUjqqJvj7QaMcCAwtrgNdahSiFofrE7o',
				},
			],
		})

		await expect(await context.execute(tx)).toPassExecution()

		expect(
			await context.state.getMultiStore().get('@bearmint/bep-067').getListHex(sender.address),
		).toStrictEqual([primaryKey])

		expect(
			BEP67MsgOp.fromBinary(
				await context.state.getMultiStore().get('@bearmint/bep-067').get(primaryKey),
			).toJson(),
		).toMatchSnapshot()
	})

	it('should attach [path] to the sender', async (context) => {
		const { primaryKey, sender, tx } = await createTx(context, {
			ops: [
				{
					type: 'path',
					value: '/ipfs/QmYjtig7VJQ6XsnUjqqJvj7QaMcCAwtrgNdahSiFofrE7o',
				},
			],
		})

		await expect(await context.execute(tx)).toPassExecution()

		expect(
			await context.state.getMultiStore().get('@bearmint/bep-067').getListHex(sender.address),
		).toStrictEqual([primaryKey])

		expect(
			BEP67MsgOp.fromBinary(
				await context.state.getMultiStore().get('@bearmint/bep-067').get(primaryKey),
			).toJson(),
		).toMatchSnapshot()
	})

	it('should attach [peerMultiaddr] to the sender', async (context) => {
		const { primaryKey, sender, tx } = await createTx(context, {
			ops: [
				{
					type: 'peerMultiaddr',
					value: '/p2p/QmcgpsyWgH8Y8ajJz1Cu72KnS5uo2Aa2LpzU7kinSoooo4',
				},
			],
		})

		await expect(await context.execute(tx)).toPassExecution()

		expect(
			await context.state.getMultiStore().get('@bearmint/bep-067').getListHex(sender.address),
		).toStrictEqual([primaryKey])

		expect(
			BEP67MsgOp.fromBinary(
				await context.state.getMultiStore().get('@bearmint/bep-067').get(primaryKey),
			).toJson(),
		).toMatchSnapshot()
	})

	it('should attach [subdomain] to the sender', async (context) => {
		const { primaryKey, sender, tx } = await createTx(context, {
			ops: [
				{
					type: 'subdomain',
					value:
						'http://bafybeie5gq4jxvzmsym6hjlwxej4rwdoxt7wadqvmmwbqi7r27fclha2va.ipfs.dweb.link',
				},
			],
		})

		await expect(await context.execute(tx)).toPassExecution()

		expect(
			await context.state.getMultiStore().get('@bearmint/bep-067').getListHex(sender.address),
		).toStrictEqual([primaryKey])

		expect(
			BEP67MsgOp.fromBinary(
				await context.state.getMultiStore().get('@bearmint/bep-067').get(primaryKey),
			).toJson(),
		).toMatchSnapshot()
	})

	it('should attach [url] to the sender', async (context) => {
		const { primaryKey, sender, tx } = await createTx(context, {
			ops: [
				{
					type: 'url',
					value: 'https://ipfs.io/ipfs/QmYjtig7VJQ6XsnUjqqJvj7QaMcCAwtrgNdahSiFofrE7o',
				},
			],
		})

		await expect(await context.execute(tx)).toPassExecution()

		expect(
			await context.state.getMultiStore().get('@bearmint/bep-067').getListHex(sender.address),
		).toStrictEqual([primaryKey])

		expect(
			BEP67MsgOp.fromBinary(
				await context.state.getMultiStore().get('@bearmint/bep-067').get(primaryKey),
			).toJson(),
		).toMatchSnapshot()
	})

	it('should attach [urlOrPath] to the sender', async (context) => {
		const { primaryKey, sender, tx } = await createTx(context, {
			ops: [
				{
					type: 'urlOrPath',
					value:
						'https://bafybeie5gq4jxvzmsym6hjlwxej4rwdoxt7wadqvmmwbqi7r27fclha2va.ipfs.dweb.link',
				},
			],
		})

		await expect(await context.execute(tx)).toPassExecution()

		expect(
			await context.state.getMultiStore().get('@bearmint/bep-067').getListHex(sender.address),
		).toStrictEqual([primaryKey])

		expect(
			BEP67MsgOp.fromBinary(
				await context.state.getMultiStore().get('@bearmint/bep-067').get(primaryKey),
			).toJson(),
		).toMatchSnapshot()
	})
})
