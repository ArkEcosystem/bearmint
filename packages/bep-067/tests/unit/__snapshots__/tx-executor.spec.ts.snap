// Vitest Snapshot v1

exports[`Tx Executor > should attach [base32cid] to the sender 1`] = `
{
  "type": "base32cid",
  "value": "bafybeie5gq4jxvzmsym6hjlwxej4rwdoxt7wadqvmmwbqi7r27fclha2va",
}
`;

exports[`Tx Executor > should attach [cid] to the sender 1`] = `
{
  "type": "cid",
  "value": "bafybeiasb5vpmaounyilfuxbd3lryvosl4yefqrfahsb2esg46q6tu6y5q",
}
`;

exports[`Tx Executor > should attach [cid] to the sender 2`] = `
{
  "type": "cid",
  "value": "QmYjtig7VJQ6XsnUjqqJvj7QaMcCAwtrgNdahSiFofrE7o",
}
`;

exports[`Tx Executor > should attach [cid] to the sender 3`] = `
{
  "type": "cid",
  "value": "zdj7WWeQ43G6JJvLWQWZpyHuAMq6uYWRjkBXFad11vE2LHhQ7",
}
`;

exports[`Tx Executor > should attach [cidPath] to the sender 1`] = `
{
  "type": "cidPath",
  "value": "QmYjtig7VJQ6XsnUjqqJvj7QaMcCAwtrgNdahSiFofrE7o/path/to/file",
}
`;

exports[`Tx Executor > should attach [ipfsPath] to the sender 1`] = `
{
  "type": "ipfsPath",
  "value": "/ipfs/QmYjtig7VJQ6XsnUjqqJvj7QaMcCAwtrgNdahSiFofrE7o",
}
`;

exports[`Tx Executor > should attach [ipfsSubdomain] to the sender 1`] = `
{
  "type": "ipfsSubdomain",
  "value": "http://bafybeie5gq4jxvzmsym6hjlwxej4rwdoxt7wadqvmmwbqi7r27fclha2va.ipfs.dweb.link",
}
`;

exports[`Tx Executor > should attach [ipfsUrl] to the sender 1`] = `
{
  "type": "ipfsUrl",
  "value": "https://ipfs.io/ipfs/QmYjtig7VJQ6XsnUjqqJvj7QaMcCAwtrgNdahSiFofrE7o",
}
`;

exports[`Tx Executor > should attach [ipnsPath] to the sender 1`] = `
{
  "type": "ipnsPath",
  "value": "/ipns/github.com",
}
`;

exports[`Tx Executor > should attach [ipnsSubdomain] to the sender 1`] = `
{
  "type": "ipnsSubdomain",
  "value": "http://bafybeiabc2xofh6tdi6vutusorpumwcikw3hf3st4ecjugo6j52f6xwc6q.ipns.dweb.link",
}
`;

exports[`Tx Executor > should attach [ipnsUrl] to the sender 1`] = `
{
  "type": "ipnsUrl",
  "value": "https://ipfs.io/ipns/github.com",
}
`;

exports[`Tx Executor > should attach [multiaddr] to the sender 1`] = `
{
  "type": "multiaddr",
  "value": "/ip4/127.0.0.1/udp/1234",
}
`;

exports[`Tx Executor > should attach [multihash] to the sender 1`] = `
{
  "type": "multihash",
  "value": "QmYjtig7VJQ6XsnUjqqJvj7QaMcCAwtrgNdahSiFofrE7o",
}
`;

exports[`Tx Executor > should attach [path] to the sender 1`] = `
{
  "type": "path",
  "value": "/ipfs/QmYjtig7VJQ6XsnUjqqJvj7QaMcCAwtrgNdahSiFofrE7o",
}
`;

exports[`Tx Executor > should attach [peerMultiaddr] to the sender 1`] = `
{
  "type": "peerMultiaddr",
  "value": "/p2p/QmcgpsyWgH8Y8ajJz1Cu72KnS5uo2Aa2LpzU7kinSoooo4",
}
`;

exports[`Tx Executor > should attach [subdomain] to the sender 1`] = `
{
  "type": "subdomain",
  "value": "http://bafybeie5gq4jxvzmsym6hjlwxej4rwdoxt7wadqvmmwbqi7r27fclha2va.ipfs.dweb.link",
}
`;

exports[`Tx Executor > should attach [url] to the sender 1`] = `
{
  "type": "url",
  "value": "https://ipfs.io/ipfs/QmYjtig7VJQ6XsnUjqqJvj7QaMcCAwtrgNdahSiFofrE7o",
}
`;

exports[`Tx Executor > should attach [urlOrPath] to the sender 1`] = `
{
  "type": "urlOrPath",
  "value": "https://bafybeie5gq4jxvzmsym6hjlwxej4rwdoxt7wadqvmmwbqi7r27fclha2va.ipfs.dweb.link",
}
`;
